consensus/src/checkpoint.rs

```diff
@@ -3,15 +3,12 @@
 
 use crate::common::{Chain, ProtocolStateChain};
 use crate::constants::SLOTS_PER_EPOCH;
+use crate::error::ConsensusError;
 use mina_crypto::base58::Base58Encodable;
 use mina_crypto::hash::{EpochSeed, StateHash};
 use mina_rs_base::protocol_state::ProtocolState;
 
-#[derive(Debug)]
-pub enum ConsensusErrTyp {
-    ConsensusInitFail,
-}
-pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusErrTyp> {
+pub fn init_checkpoints(genesis: &mut ProtocolState) -> Result<(), ConsensusError> {
```

Comment:
How about creating a Checkpoint trait and making this a trait method? In general should we prefer OO way in our api design or non-OO way?

Reply:
Not sure how we could possibly enforce OO around every code, but definitely trait based APIs wherever possible (the rusty way). Yes am thinking of refactoring the checkpoint related parts, as part of this PR.

Reply:
I notice the `init_checkpoint` function is only being used in test cases at the moment, so i have moved it to under `tests/checkpoint.rs` within `mod tests {}`. There isn't really much to refactor in checkpoint.rs related to code design at the moment as we simply have a `is_short_range` function there. Based on how it's being used in `select_secure_chain` API, it feels having `is_short_range` as a free helper function is appropriate.

Reply:
Sounds good, thx!